script_tools/upgrade_agent.sh.tpl

```diff
@@ -3,6 +3,24 @@ OS_INFO=""
 OS_TYPE=""
 RC_LOCAL_FILE=/etc/rc.d/rc.local
 
+get_cpu_arch () {{
+    local cmd=$1
+    CPU_ARCH=$($cmd)
+    CPU_ARCH=$(echo $CPU_ARCH | tr 'A-Z' 'a-z')
+    if [[ "$CPU_ARCH" =~ "x86_64" ]]; then
+        return 0
+    elif [[ "$CPU_ARCH" =~ "x86" || "$CPU_ARCH" =~ ^i[3456]86 ]]; then
+        CPU_ARCH="x86"
+        return 0
+    elif [[ "$CPU_ARCH" =~ "aarch" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}}
+
+get_cpu_arch "uname -p" || get_cpu_arch "uname -m"  || arch || echo get_cpu_arch "Failed to get CPU arch, please contact the developer."
```

Comment:
get_cpu_arch "uname -p" || get_cpu_arch "uname -m" || arch || fail get_cpu_arch FAILED "Failed to get CPU arch, please contact the developer."